src/BizHawk.Client.EmuHawk/tools/Lua/Libraries/TAStudioLuaLibrary.cs

```diff
@@ -143,30 +143,32 @@ public void SetPlayback(object frame)
 					throw new InvalidOperationException("tastudio.setplayback() is not allowed during input/memory callbacks");
 				}
 
-				_luaLibsImpl.IsUpdateSupressed = true;
 
 				int f;
 				if (frame is long frameNumber)
 				{
 					f = (int)frameNumber;
 				}
+				else if (frame is double frameNumber2)
+				{
+					f = (int)frameNumber2;
+				}
 				else
 				{
-					f = Tastudio.CurrentTasMovie.Markers.FindIndex((string)frame);
-					if (f == -1)
+					f = Tastudio.CurrentTasMovie.Markers.FindIndex((string) frame);
+					if (f != -1)

```

Comment:
`>= 0`, and I think this should be using a separate temp variable rather than reassigning `f`.

Reply:
I disagree with `>= 0`. `FindIndex` specifies that it returns -1 if no match is found. Checking for that specific value makes it clear what the code intends to do. And if we don't want to trust `FindIndex` to work correctly then we should be checking for the upper bound of valid indexes too.